### PR TITLE
Pa mapping

### PR DESCRIPTION
**Pull request by mdellabitta**
DO NOT MERGE

Looking for a code review of my implementation of the PA mapping.

The MAP validator is complaining about their not being an `object` property, but the mapping spreadsheet says 'none' for that property. 

DO NOT MERGE

### CODE REVIEW

**Comment by markbreedlove** [comment]
Hey, @mdellabitta, `object` is the thumbnail image, and it's an optional property. I don't recall what the validation messages look like, but were these warnings or errors?

I'll have more comments tomorrow.  At a glance, it looks good, but I haven't read it closely.

**Comment by mdellabitta** [comment]
The message is something like "Object is a required property."

**Comment by moltude** [comment]
@mdellabitta The sheet may be out of date because object/preview should be the last value in dc:identifier. Perhaps @guegueng can clarify? 

https://github.com/dpla/heidrun-mappings/blob/master/heidrun/pa_dc.rb#L101-L103

**Comment by markbreedlove** [comment]
For what it's worth, the MAPv4 schema document at http://dp.la/info/wp-content/uploads/2015/03/MAPv4.pdf says that the Obligation for Object is "0-1."

**Comment by mdellabitta** [comment]
OK, I'll just ignore that, then!

I've resolved the merge conflicts by rebasing over develop.

**Comment by markbreedlove** [comment]
Ah, but here's the catch.  In the MAPv3.1 schema the obligation is 1.  See https://dp.la/info/wp-content/uploads/2013/04/DPLA-MAP-V3.1-2.pdf

**Comment by markbreedlove** [comment]
(And ingestion 1 is using MAPv3.1.)

**Comment by guegueng** [comment]
Just to explain, in MAP 3.1 we were putting the thumbnail link in "object" which is actually an incorrect usage of the property. We fixed it with MAP 4.0, but, as Mark noted, Ingestion 1 still uses MAP 3.1

File: lib/create_mapper.py [hunk 1]
```diff
@@ -124,7 +128,9 @@ def _create_mi_mapper(data):
         'mdl_json':     lambda d: _create_mdl_json_mapper(d),
         'cdl_json':     lambda d: _create_cdl_json_mapper(d),
         'in':           lambda d: _create_in_mapper(d),
-        'mi':           lambda d: _create_mi_mapper(d)
+        'mi':           lambda d: _create_mi_mapper(d),
+        'pa':           lambda d: _create_pa_mapper(d),
+        'pa':           lambda d: _create_pa_mapper(d)
```
Comment (markbreedlove) [inline lib/create_mapper.py]: Looks like this is a duplicate line.

File: lib/mappers/pa_mapper.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    def map_relation(self):
+        prop = "relation"
+        if exists(self.provider_data, prop):
+            relations = iterify(self.provider_data.get(prop))
+            self.update_source_resource({"isPartOf": relations[0]})
+            if len(relations) > 1:
+                self.update_source_resource({"relation": relations[1:]})
+
+    def map_type(self):
```
Comment (moltude) [inline lib/mappers/pa_mapper.py]: @mdellabitta Purely as an FYI but there is an alternative way to do this with the `enrich-type` enrichment. https://github.com/dpla/ingestion/blob/e0778e6afdcbf2eba35a2b3fefbf52dda6fdea97/lib/akamod/enrich-type.py

**Comment by moltude** [comment]
Do not merge until QA is done

**Comment by markbreedlove** [comment]
FYI, this looks fine to me. I don't know if anyone's been waiting for further input from me, but I don't have much else to add.

In general, regarding this and other mappings, I just think we might start adding docstrings to say how an overridden mapping method is different than the parent class's.

**Comment by moltude** [comment]
Closing this PR until the work is complete

**Pull request closed without merging**